core/src/main/java/edu/wpi/grip/core/metrics/Analysis.java

```diff
@@ -0,0 +1,90 @@
+package edu.wpi.grip.core.metrics;
+
+import com.google.common.base.MoreObjects;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Analysis data of a timed action. Contains
+ * <ul>
+ * <li>The number of samples taken</li>
+ * <li>The average of the samples taken. This may not be an arithmetic mean, or even an average
+ * of <i>all</i> the samples</li>
+ * <li>Statistical analysis of the data.</li>
+ * </ul>
+ */
+public class Analysis {
+
+  // <0: error
+  // 0 = no change from original state
+  // 1 = returns most recent input
+  // >1: error
+  private static final double alpha = 0.35;
+
+  private final ExponentialMovingAverage ema = new ExponentialMovingAverage(alpha);
```

Comment:
Maybe change this to a normal arithmetic moving average, the output of this still isn't very consistent.